crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -241,7 +244,26 @@ impl<'a> ReFunc<'a> {
             // pattern in string
             ReFuncKind::Search => Some(self.compare_expr(CmpOp::In)),
             // string == pattern
-            ReFuncKind::Fullmatch => Some(self.compare_expr(CmpOp::Eq)),
+            ReFuncKind::Fullmatch => {
+                let old_expr = self.compare_expr(CmpOp::Eq);
+                // Make it string == pattern instead of pattern == string
+                if let Expr::Compare(ExprCompare {
+                    left,
+                    ops,
+                    comparators,
+                    ..
+                }) = old_expr
+                {
+                    Some(Expr::Compare(ExprCompare {
+                        range: TextRange::default(),
+                        left: Box::new(comparators[0].clone()),
+                        ops,
+                        comparators: Box::new([*left]),
+                    }))
+                } else {
+                    None
+                }
```

Comment:
I think you can just in-line `self.compare_expr` here and swap `left` and `comparators`. That seems easier than calling `compare_expr` and unpacking the `Expr` again. This is basically what you have in your `Some(Expr::Compare...` but saves you from having to call `self.compare_expr` and then `if let`.

After that, I think you could delete the comment too since the implementation will now be consistent with the `string == pattern` comment above. It looks like I had the right idea with that comment but implemented it backwards :sweat_smile: 

You may not want to apply this change through GitHub because I think the indentation is wrong, but I thought an example might help.

```suggestion
                    Some(Expr::Compare(ExprCompare {
                        range: TextRange::default(),
                        left: Box::new(self.string.clone()),
                        ops,
                        comparators: Box::new([self.pattern.clone()]),
                    }))
```

Reply:
This is nice. I overlooked this way of implementation. Corrected in my recent commit.